server/lib/availabilitySync.ts

```diff
@@ -234,62 +183,115 @@ class AvailabilitySync {
     const mediaRepository = getRepository(Media);
     const requestRepository = getRepository(MediaRequest);
 
-    const isTVType = media.mediaType === 'tv';
-
     try {
-      const request = await requestRepository.findOne({
-        relations: {
-          media: true,
-        },
-        where: { media: { id: media.id }, is4k: is4k ? true : false },
-      });
-
       logger.info(
-        `Media ID ${media.id} does not exist in your ${
+        `Media with TMDB ID ${media.tmdbId} does not exist in your ${
           is4k ? '4k' : 'non-4k'
         } ${
-          isTVType ? 'Sonarr' : 'Radarr'
+          media.mediaType === 'tv' ? 'Sonarr' : 'Radarr'
         } and Plex instance. Status will be changed to unknown.`,
         { label: 'AvailabilitySync' }
       );
 
+      const requests = await requestRepository.find({
+        relations: {
+          media: true,
+        },
+        where: { media: { id: media.id }, is4k: is4k },
+      });
+
+      // Check if a season is processing or pending to
+      // make sure we set media to the correct status
+      let mediaStatus = MediaStatus.UNKNOWN;
+
+      if (media.mediaType === 'tv') {
+        mediaStatus = await this.findMediaStatus(requests);
+      }
+
       await mediaRepository.update(
         media.id,
         is4k
           ? {
-              status4k: MediaStatus.UNKNOWN,
-              serviceId4k: null,
-              externalServiceId4k: null,
-              externalServiceSlug4k: null,
-              ratingKey4k: null,
+              status4k: mediaStatus,
+              serviceId4k:
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.serviceId4k
+                  : null,
+              externalServiceId4k:
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceId4k
+                  : null,
+              externalServiceSlug4k:
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceSlug4k
+                  : null,
+              ratingKey4k:
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.ratingKey4k
+                  : null,
             }
           : {
-              status: MediaStatus.UNKNOWN,
-              serviceId: null,
-              externalServiceId: null,
-              externalServiceSlug: null,
-              ratingKey: null,
+              status: mediaStatus,
+              serviceId:
+                mediaStatus === MediaStatus.PROCESSING ? media.serviceId : null,
+              externalServiceId:
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceId
+                  : null,
+              externalServiceSlug:
+                mediaStatus === MediaStatus.PROCESSING
+                  ? media.externalServiceSlug
+                  : null,
+              ratingKey:
+                mediaStatus === MediaStatus.PROCESSING ? media.ratingKey : null,
```

Comment:
Same thing we are doing before which is to make sure we are setting the media to the correct status if requests remain.